backend/src/controllers/review.controller.ts

```diff
@@ -148,7 +148,7 @@ Consider the further instructions and example replies if provided by the manager
     this.debug(`Sentiment over time initialized for ${sentimentOverTime.size} sentiments`);
 
     reviews.forEach((review) => {
-      const dateData = sentimentOverTime.get(review.date.toLocaleDateString())!;
+      const dateData = sentimentOverTime.get(new Date(review.date).toLocaleDateString())!;
```

Comment:
Bug: It didnt work because `review.date` is either a date or string
Solution: Wrap with `new Date()`